src/dvc_data/hashfile/state.py

```diff
@@ -109,23 +153,23 @@ def get(self, path, fs):
             return None, None
 
         raw = self.hashes.get(path)
-        if not raw:
-            return None, None
-
-        try:
-            entry = json.loads(raw)
-        except ValueError:
+        entry = self._get(path, fs, raw)
+        if not entry:
             return None, None
+        return Meta(size=entry["size"]), HashInfo.from_dict(entry["hash_info"])
 
-        try:
-            actual = fs.checksum(path)
-        except FileNotFoundError:
-            return None, None
+    def get_many(self, paths, fs):
+        from .meta import Meta
 
-        if entry["checksum"] != actual:
-            return None, None
+        if not isinstance(fs, LocalFileSystem):
+            return
 
-        return Meta(size=entry["size"]), HashInfo.from_dict(entry["hash_info"])
+        for path, value in self.hashes.get_many(paths):
+            meta, hash_info = None, None
+            if entry := self._get(path, fs, value):
```

Comment:
I see it is a walrus season apparently 😄 Let's not overcomplicate it.
```suggestion
            entry = self._get(path, fs, value)
            if entry:
```